### ISSUE TITLE
Members of derived classes are implicitly added to interfaces

### ISSUE THREAD

**Issue by SLaks** [google/closure-compiler#657]
I have a class which extends an interface and adds a method to it.

The type checker doesn't complain when I call that method on a variable declared as the interface:

``` js
// ==ClosureCompiler==
// @compilation_level ADVANCED_OPTIMIZATIONS
// ==/ClosureCompiler==

/** @interface */
function Base() {}

/**
 * @implements {Base}
 * @constructor
 */
function SubClass() {}
SubClass.prototype.subMethod = function() { };

/** @type {Base} */
var b;

b.subMethod();  // No error - why not?
b.doesntExist(); // Errors as expected

/** @type {string} */
var a = b;  // Also errors, proving that b is of type Base, not SubClass
```
